### PR TITLE
ISPN-12446 Including configuration examples in the server distribution

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12446

### CODE REVIEW

**Comment by tristantarrant** [comment]
Rebased, squashed and repushed

**Comment by ryanemerson** [comment]
Overall I like this a lot. The only example that I think could confuse the user is`AuthenticationImplicit.xml`, for me it's not clear what this does from the name alone.

File: server/runtime/src/main/server/docs/examples/README.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[id='server_configuration_examples']
```
Comment (pruivo) [inline server/runtime/src/main/server/docs/examples/README.adoc:1]: should be a text file IMO

**Comment by pruivo** [comment]
xsite example has no references to RELAY2. Also, it doesn't work.
```
[pedro@pedro-desktop infinispan-server-12.0.0-SNAPSHOT]$ cp docs/examples/XSiteServer.xml server/conf/
[pedro@pedro-desktop infinispan-server-12.0.0-SNAPSHOT]$ ./bin/server.sh -c XSiteServer.xml
10:30:02,784 INFO  (main) [BOOT] JVM OpenJDK 64-Bit Server VM Oracle Corporation 11.0.7+10
10:30:02,794 INFO  (main) [BOOT] JVM arguments = [-server, -Xlog:gc*:file=/home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT/server/log/gc.log:time,uptimemillis:filecount=5,filesize=3M, -Xms64m, -Xmx512m, -XX:MetaspaceSize=64M, -Djava.net.preferIPv4Stack=true, -Djava.awt.headless=true, -Dvisualvm.display.name=infinispan-server, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dinfinispan.server.home.path=/home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT, -classpath, :/home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT/boot/infinispan-server-runtime-12.0.0-SNAPSHOT-loader.jar, org.infinispan.server.loader.Loader, org.infinispan.server.Bootstrap, -c, XSiteServer.xml]
10:30:02,794 INFO  (main) [BOOT] PID = 971327
10:30:02,825 INFO  (main) [org.infinispan.SERVER] ISPN080000: Infinispan Server starting
10:30:02,826 INFO  (main) [org.infinispan.SERVER] ISPN080017: Server configuration: /home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT/server/conf/XSiteServer.xml
10:30:02,826 INFO  (main) [org.infinispan.SERVER] ISPN080032: Logging configuration: /home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT/server/conf/log4j2.xml
10:30:03,263 WARN  (main) [org.infinispan.commons.util.StringPropertyReplacer] ISPN000901: Property infinispan.cluster.stack could not be replaced as intended!
10:30:03,266 FATAL (main) [org.infinispan.SERVER] ISPN080028: Infinispan Server failed to start org.infinispan.commons.CacheConfigurationException: ISPN000540: No such JGroups stack '${infinispan.cluster.stack}'
	at org.infinispan.configuration.parsing.Parser.parseTransport(Parser.java:986)
	at org.infinispan.configuration.parsing.Parser.parseContainer(Parser.java:697)
	at org.infinispan.configuration.parsing.Parser.readElement(Parser.java:120)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:224)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:194)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:180)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:169)
	at org.infinispan.server.Server.parseConfiguration(Server.java:280)
	at org.infinispan.server.Server.<init>(Server.java:210)
	at org.infinispan.server.Bootstrap.runInternal(Bootstrap.java:157)
	at org.infinispan.server.tool.Main.run(Main.java:98)
	at org.infinispan.server.Bootstrap.main(Bootstrap.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.infinispan.server.loader.Loader.run(Loader.java:76)
	at org.infinispan.server.loader.Loader.main(Loader.java:39)

10:30:03,281 INFO  (Thread-0) [org.infinispan.SERVER] ISPN080002: Infinispan Server stopping
10:30:03,283 INFO  (Thread-0) [org.infinispan.SERVER] ISPN080003: Infinispan Server stopped
```

File: server/runtime/src/main/server/docs/examples/README.adoc (hunk 1 above)
Comment (tristantarrant) [reply server/runtime/src/main/server/docs/examples/README.adoc:1]: I'll leave that to @oraNod

**Comment by tristantarrant** [comment]
@pruivo I have changed the XSiteServer.xml config. It should work OOTB now

**Comment by pruivo** [comment]
`infinispan-xsite.xml` could be removed from `server/conf` IMO.

**Comment by pruivo** [comment]
@tristantarrant are the examples supposed to work out of the box?

**Comment by pruivo** [comment]
`AuthenticationImplicit.xml` fails with `Caused by: java.io.FileNotFoundException: /home/pedro/Work/infinispan/server/runtime/target/infinispan-server-12.0.0-SNAPSHOT/server/conf/alternate-users.properties (No such file or directory)`

rename `Authentication.xml` => `AuthenticationExplicity.xml`. Also, it should `AuthenticationImplicit.xml` + explicity authentication mechanisms in connectors.

**Comment by pruivo** [comment]
`AuthorizationProperties.xml` defines a cache with cross-site but without cross-site jgroups stacks.

**Comment by pruivo** [comment]
remove `ClusteredServer.xml`? doesn't add anything useful IMO.

**Comment by pruivo** [comment]
`CustomStore.xml` & `ClusteredServer.xml` are the same file

**Comment by tristantarrant** [comment]
AuthenticationImplicit.xml fails: creating those files is an exercise left to the user
Note that the AuthenticationKerberos.xml config needs a keytab we cannot provide

**Comment by tristantarrant** [comment]
@pruivo I'm selecting only a subset of the config files now.

File: server/runtime/src/main/server/docs/examples/README.adoc (hunk 1 above)
Comment (oraNod) [reply server/runtime/src/main/server/docs/examples/README.adoc:1]: fair enough @pruivo I'll make some changes.

File: server/runtime/build.xml [hunk 2]
```diff
@@ -86,6 +86,25 @@
             <include name="xstream-bsd.txt"/>
          </fileset>
       </copy>
+      <pathconvert property="example.files" pathsep=" ">
+         <fileset dir="${examples.dir}">
+            <include name="AuthenticationExplicitTest.xml"/>
+            <include name="AuthenticationKerberosTest.xml"/>
+            <include name="AuthenticationKeyCloakTest.xml"/>
+            <include name="AuthenticationLDAPTest.xml"/>
+            <include name="AuthenticationServerMultipleEndpointsTest.xml"/>
```
Comment (pruivo) [inline server/runtime/build.xml:95]: this file doesn't exist. It doesn't have "Test" at the end.

File: server/runtime/src/main/server/docs/examples/README.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+This folder contains configuration examples for different environments.
+Open each example configuration with any text editor to view inline comments that describe the configuration.
+
+To start the server with one of the configuration examples, do the following:
+
+. Copy the configuration file to the `$SERVER_HOME/conf` directory.
+. Specify the filename with the `-c` argument when you start the server.
+
+For example, start the server with a configuration file named "Authentication.xml":
```
Comment (pruivo) [inline server/runtime/src/main/server/docs/examples/README.adoc:11]: choose another file. `Authentication.xml` doesn't exist anymore.

**Review by pruivo — CHANGES_REQUESTED** [review]

**Comment by pruivo** [comment]
I wouldn't use `AuthenticationServerMultipleEndpointsTest.xml` but create a configuration from scratch with Wolf "request" closer as possible from "default" infinispan.xml:

* 2 different interfaces, public & management
* 2 security realms, user & management
* 2 socket bindings, public & management
* 2 endpoints, public HR/Rest (anonym metrics?) & management Rest

ps. I'm not a big fan of "alternate ports" for configs where it isn't required. IMO, It adds verbosity to the config without any value-added.
```
   <!-- Additional bindings on different ports. -->
   <socket-binding name="alternate-1" port="11223"/>
   <socket-binding name="alternate-2" port="11224"/>
   <socket-binding name="alternate-3" port="11225"/>
   <socket-binding name="alternate-4" port="11226"/>
```

**Comment by tristantarrant** [comment]
Closing this: we need to have a better curated list of examples.

**Pull request closed without merging**